module/x/peggy/keeper/batch.go

```diff
@@ -166,7 +159,7 @@ func (k Keeper) CancelOutgoingTXBatch(ctx sdk.Context, tokenContract string, non
 	}
 	for _, tx := range batch.Transactions {
 		tx.Erc20Fee.Contract = tokenContract
-		k.prependToUnbatchedTXIndex(ctx, tokenContract, tx.Erc20Fee.PeggyCoin(), tx.Id)
+		k.prependToUnbatchedTXIndex(ctx, tokenContract, *tx.Erc20Fee, tx.Id)
```

Comment:
can `*tx.Erc20Fee` be nil? A test case to check this should be made. 

Quickly looking at the code it seems like a nil outgoingtx could be submitted.

Reply:
Good catch. I've gotten out of the habit of actually checking pointers since that's not something I have to concern myself with in Rust.

Reply:
huh this really wasn't introduced in this pr, the possibility existed previously just less explicitly. We where still calling PeggyCoin() on a reference.